sia/src/types/common.rs

```diff
@@ -132,6 +183,66 @@ impl fmt::Display for ChainIndex {
     }
 }
 
+/// A Block is a collection of transactions
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Block {
```

Comment:
Similar thing. Would be nice to test the encoding of a block.

Reply:
I added a test for the v1 encoding and removed the v2 encoding because it's significantly more involved and this is already a giant PR. Will add the v2 encoding back in a follow up.